FirebaseSegmentation/Sources/SEGContentManager.m

```diff
@@ -63,69 +63,112 @@ - (instancetype)initWithDatabaseManager:databaseManager networkManager:networkMa
   return self;
 }
 
-// TODO(dmandar) IID only supports default instance. Modify for FIS.
-- (FIRInstanceID *)instanceIDForApp:(NSString *)firebaseApp {
-  return [FIRInstanceID instanceID];
+- (FIRInstallations *)installationForApp:(NSString *)firebaseApp {
+  return [FIRInstallations installationsWithApp:[FIRApp appNamed:firebaseApp]];
 }
 
 - (void)associateCustomInstallationIdentiferNamed:(NSString *)customInstallationID
                                       firebaseApp:(NSString *)firebaseApp
                                        completion:(SEGRequestCompletion)completionHandler {
-  // Get the latest instance identifier
-  if (![self instanceIDForApp:firebaseApp]) {
-    completionHandler(NO, @{kErrorDescription : @"InstanceID SDK not available"});
+  // Get the latest installation identifier
+  FIRInstallations *installation = [self installationForApp:firebaseApp];
+  if (installation == nil) {
+    completionHandler(NO, @{kErrorDescription : @"Firebase Installations SDK not available"});
   }
   __weak SEGContentManager *weakSelf = self;
-  [[FIRInstanceID instanceID] instanceIDWithHandler:^(FIRInstanceIDResult *_Nullable result,
-                                                      NSError *_Nullable error) {
+  [installation
+      installationIDWithCompletion:^(NSString *__nullable identifier, NSError *_Nullable error) {
```

Comment:
nit: please use either `__nullable` or `_Nullable`. They are interchangeable but `_Nullable` seems to be wider used in Firebase codebase.